### PR TITLE
webcam at dagster pipeline

### PR DESCRIPTION
**Pull request by the-infinity**
Sets some env vars and mounts for https://github.com/mobidata-bw/ipl-dagster-pipeline/pull/177 . Requires https://github.com/mobidata-bw/ipl-ansible/pull/77

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -130,3 +130,7 @@ GOACCESS_KEEP_LAST_DAYS=60
 # SFTP
 SFTP_IMAGE=atmoz/sftp:alpine-3.7
 SFTP_PORT=10022
+
+# Webcam
+WEBCAM_KEEP_DAYS=7
```
Comment (derhuerst) [inline .env:135]: Let's add an inline explanation for these variables.

```suggestion
# How many days (including the current) should be mirrored.
WEBCAM_KEEP_DAYS=7
```

File: .env.local.example [hunk 2]
```diff
@@ -99,3 +99,8 @@ RADVIS_WFS_PASSWORD=secret
 # SVZBW DATEX2 Download URL
 ROADWORKS_SVZBW_DATEX2_DOWNLOAD_URL=https://url
 
+# Webcam
+WEBCAM_FTP_SERVER=server
+WEBCAM_FTP_USER=user
+WEBCAM_FTP_PASSWORD=password
+WEBCAM_BASIC_AUTH=secret
```
Comment (derhuerst) [inline .env.local.example:106]: Please add an explanation what this environment variable does and what its format looks like.

File: docker-compose.yml [hunk 3]
```diff
@@ -160,14 +160,19 @@ services:
       - ./etc/index-page:/var/www/index-page:ro
       # public IPL assets, e.g. traffic data (served on :80)
       - ./var/www/datasets:/var/www/datasets:ro
+      # password protected webcam data (served on :80)
```
Comment (derhuerst) [inline docker-compose.yml:163]: Let's make it clear that the password protection happens in front of `caddy`.

```suggestion
      # webcam data (served on :80, password-protected by the `ingress` service)
```

File: docker-compose.yml [hunk 4]
```diff
@@ -414,13 +419,21 @@ services:
       - RADVIS_WFS_PASSWORD=${RADVIS_WFS_PASSWORD:?missing/empty}
       # Roadworks download url
       - ROADWORKS_SVZBW_DATEX2_DOWNLOAD_URL=${ROADWORKS_SVZBW_DATEX2_DOWNLOAD_URL:?missing/empty}
+      # Webcam settings
+      - IPL_WEBCAM_SERVER=${WEBCAM_FTP_SERVER:?missing/empty}
```
Comment (derhuerst) [inline docker-compose.yml:423]: [nitpick]

Unless there's a good reason, I'd prefer having the same naming in `.env` and within the container(s).

(The same applies to the variables below.)

**Review by derhuerst — CHANGES_REQUESTED** [review]
Not sure if this is the case right now, but I would like `ipl-orchestration` to be configured so that it works out-of-the-box, i.e. the webcam mirroring being turned off based on an environment variable (e.g. if `$WEBCAM_SERVER` is empty?).

---

Also, the changes should be added to the changelog, ideally in each respective commit.

File: .env.local.example (hunk 2 above)
Comment (the-infinity) [reply .env.local.example:106]: Done

File: .env (hunk 1 above)
Comment (the-infinity) [reply .env:135]: Added

File: docker-compose.yml (hunk 3 above)
Comment (the-infinity) [reply docker-compose.yml:163]: Added

File: docker-compose.yml (hunk 4 above)
Comment (the-infinity) [reply docker-compose.yml:423]: I would actually prefer it this way: in context of the env var, the IPL prefix does not make much sense, in context of the container, it does.

**Review by derhuerst — APPROVED** [review]
Looks good!

Can you squash your review commits into the original ones?

**Pull request merged**